src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingLeadingAsteriskCheck.java

```diff
@@ -0,0 +1,127 @@
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks if the javadoc has leading asterisk on each line or not.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingLeadingAsterisk"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * &#47;** // violation, javadoc has lines without leading asterisk.
+ *  * Some description here.
+ *    Another line of description.
+ *  *&#47;
+ * public class Test {
+ *   &#47;** // violation, javadoc has lines without leading asterisk.
+ *    * Some description here.
+ *    * Another line of description.
+ *      Another line of description.
+ *    *&#47;
+ *   public void test() {}
+ *
+ *   &#47;** Some description here. *&#47; // OK
+ *   public void test1() {}
+ *
+ *   &#47;** // OK
+ *    * Some description here.
+ *    * Another line of description.
+ *    *&#47;
+ *   public void test2() {}
+ *
+ *   &#47;** // violation, javadoc has lines without leading asterisk.
+ *    * Some description here.
+ *
+ *    * Another line of description.
+ *    *&#47;
+ *   public void test3() {}
+ * }
+ * </pre>
+ *
+ * @since 8.32
+ */
+@StatelessCheck
+public class MissingLeadingAsteriskCheck extends AbstractCheck {
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_MISSING_ASTERISK = "javadoc.missing.asterisk";
+
+    /**
+     * The new line character that separates each line of javadoc.
+     */
+    private static final Pattern MATCH_COMMENT_SEPARATOR = CommonUtil.createPattern("\r\n?|\n");
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.BLOCK_COMMENT_BEGIN,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final String commentContent = JavadocUtil.getBlockCommentContent(ast);
+        if (JavadocUtil.isJavadocComment(commentContent)) {
+            final String[] commentLines = MATCH_COMMENT_SEPARATOR.split(commentContent);
+            // We need to ignore the last string.
+            // As a java doc always has a empty string at the end, which is not a violation.
```

Comment:
Also, one-line Javadoc comments will be ignored here.
```
/** * Javadoc with a leading asterisk on the first line is legal, but weird. */
```
But what about [Scala-style](https://docs.scala-lang.org/style/scaladoc.html) (compact) comments:
```
/** First line
  * Second line
  * Third line
  */
```
?
And this one:
```
/** First line
  * Second line
    Third line  **/ // violation
```
?
 Please add them as test cases.

Reply:
> ```
> /** First line
>   * Second line
>   * Third line
>   */
> ```


This is okay, it will not generate any violations as this is a valid comment.

> ```
> /** First line
>   * Second line
>     Third line  **/ // violation
> ```

Yes, this one will generate a violation. I'll add them to the test case input.

> Also, one-line Javadoc comments will be ignored here.

We ignored inline Javadoc comments, you can see the details in this [conversation](https://github.com/checkstyle/checkstyle/pull/8085#discussion_r408239265)

Reply:
Added the given cases to UT input.

Reply:
> Added the given cases to UT input.

I still see no cases like [this one](http://hg.openjdk.java.net/jdk10/master/file/be620a591379/src/java.base/share/classes/jdk/internal/reflect/ClassFileAssembler.java#l584)

Reply:
I added this test case to the input. But it didn't generate a violation, due to this code
```
// We need to ignore the last string.
// As a java doc always has an empty string at the end, which is not a violation.
 for (int i = 0; i < commentLines.length - 1; i++) {
```
To fix this, I was thinking to check the last string only if `BLOCK_COMMENT_END` and that comment are on the same line. What do you think? Is this okay?

Reply:
Yes, this will work.

Reply:
There is just one problem. The whole comment is a single ASTNode. So I cannot use `TokenUtil.areOnSameLine`. I'll need to manually compare the line numbers.

Reply:
@Gaurav-Punjabi, is it possible to ignore the last line if it contains only whitespaces?

Reply:
Let me check.

Reply:
@pbludov I don't think to verify the last line with a whitespace would be reliable. I'm saying this because not all the Javadoc end with a white space. Some end with a blank string or new line. 
##### Example 1:
```
/** // violation
   ** OK
   *
       violation
   violation

   **/
```
When this example is parsed it ends with a blank string.

##### Example 2:
Where as in the example below:
```
  /**
   * Some description here.
   * @param a description of the parameter.
   */
```
The last line is a single space.

Reply:
I would recommend to just compare lineNo of last line and `BLOCK_COMMENT_END` after the for loop ends. If they are on the same line, we will further validate it otherwise we'll just ignore it. I think this will work in all the case without any issues.

Reply:
@pbludov It worked! I have updated the code and added the given case to the UT Input. Please review, I'll provide a new regression report for this as the code has been updated.